mesonbuild/environment.py

```diff
@@ -939,10 +947,17 @@ def sanitize(p):
                     cls = GnuCCompiler if lang == 'c' else GnuCPPCompiler
 
                 linker = self._guess_nix_linker(compiler, cls, for_machine)
-                return cls(
-                    ccache + compiler, version, for_machine, is_cross,
-                    info, exe_wrap, defines, full_version=full_version,
-                    linker=linker)
+
+                if lang == 'c':  # There's no 'defines' anymore in C Compiler classes.
+                    return cls(
+                        ccache + compiler, version, for_machine, is_cross,
+                        info, exe_wrap, full_version=full_version,
+                        linker=linker)
```

Comment:
I don't understand this change, the GnuCCompiler still has a `defines` member, it looks like you've only take it out of the ElbrusCCompiler, or am I missing something?

Reply:
I have a gut feeling that this may be what's causing the msys failure, no proof, just a gut feeling.

Reply:
Oh, you're right. I notices such a fact in CCompiler, but did not figured out that ElbrusCCompiler is derived from GnuCCompiler, not CCompiler, and even if most of other ...CCompiler classes dropped this argument, I have no need to do the same, as GnuCCompiler still has it and I may pass it there.

Fixed it back. Hope checks will pass successfully (unit tests both on x86_64 and Elbrus passed).